### PR TITLE
XW-2836 | displaying related video on chosen articles

### PR DESCRIPTION
**Pull request by weronika**
https://wikia-inc.atlassian.net/browse/XW-2836
https://wikia-inc.atlassian.net/browse/XW-2862

### CODE REVIEW

File: extensions/wikia/ArticleVideo/ArticleVideo.hooks.php [hunk 1]
```diff
@@ -27,6 +36,20 @@ public static function onMakeGlobalVariablesScript( array &$vars, OutputPage $ou
 			];
 		}
 
+		if ( isset( $wg->articleVideoRelatedVideos ) ) {
+			$relatedVideo = ArticleVideoController::getRelatedVideoData( $wg->articleVideoRelatedVideos, $title );
+
+			if ( isset( $relatedVideo['videoId'] ) ) {
+				$vars['wgArticleRelatedVideoData'] = [
+					'videoId' => $relatedVideo['videoId'],
+					'playerParams' => [
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/ArticleVideo.hooks.php]: I don't think it fits `wgArticleVideoData` as it's not really _a video data_ but rather _the player config_. Also it just duplicates the one from `wgArticleVideoData`.

File: extensions/wikia/ArticleVideo/ArticleVideoController.class.php [hunk 2]
```diff
@@ -13,4 +13,22 @@ public function index() {
...
+		$wg = F::app()->wg;
+
+		$title = RequestContext::getMain()->getTitle()->getPrefixedDBkey();
+
+		$this->setVal( 'relatedVideo', self::getRelatedVideoData( $wg->articleVideoRelatedVideos, $title) );
+	}
+
+	public static function getRelatedVideoData( $relatedVideos, $title ) {
+		foreach ( $relatedVideos as $videoData ) {
+			if ( isset($videoData['articles']) && in_array( $title, $videoData['articles'] ) ) {
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/ArticleVideoController.class.php]: Whitespaces are missing here.

File: extensions/wikia/ArticleVideo/ArticleVideoController.class.php [hunk 3]
```diff
@@ -13,4 +13,22 @@ public function index() {
 			$wg->extensionsPath . '/wikia/ArticleVideo/images/close.svg' );
 		$this->setVal( 'videoDetails', $wg->articleVideoFeaturedVideos[$title] );
 	}
+
+	public function related() {
+		$wg = F::app()->wg;
+
+		$title = RequestContext::getMain()->getTitle()->getPrefixedDBkey();
+
+		$this->setVal( 'relatedVideo', self::getRelatedVideoData( $wg->articleVideoRelatedVideos, $title) );
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/ArticleVideoController.class.php]: Space is missing.

File: extensions/wikia/ArticleVideo/ArticleVideoController.class.php [hunk 4]
```diff
@@ -13,4 +13,22 @@ public function index() {
 			$wg->extensionsPath . '/wikia/ArticleVideo/images/close.svg' );
 		$this->setVal( 'videoDetails', $wg->articleVideoFeaturedVideos[$title] );
 	}
+
+	public function related() {
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/ArticleVideoController.class.php]: If this is called `related()`, should the one above be `featured()`?

File: extensions/wikia/ArticleVideo/scripts/article-related-video.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+			ooyalaVideoElementId = 'ooyala-article-related-video',
+			track = tracker.buildTrackingFunction({
+				category: 'article-video',
+				trackingMethod: 'analytics'
+			});
+
+		function initVideo(ooyalaContainerId, videoId, onCreate) {
+			var playerParams = window.wgArticleRelatedVideoData.playerParams;
+
+			OoyalaPlayer.initHTMl5Players(ooyalaContainerId, playerParams, videoId, onCreate);
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/scripts/article-related-video.js]: Please merge the changes from XW-2814, this method is renamed.

File: extensions/wikia/ArticleVideo/scripts/article-related-video.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+		}
+
+		function placeRelatedVideo(player) {
+			var $articleContent = $('#mw-content-text'), placements, $followingSibling, rating = 0;
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/scripts/article-related-video.js]: The convention is to have each variable in a separate line, defined ones first.

File: extensions/wikia/AssetsManager/config.php [hunk 7]
```diff
@@ -3172,23 +3172,44 @@
...
+];
+
+$config['article_related_video_js'] = [
+	'type' => AssetsManager::TYPE_JS,
+	'skin' => [ 'oasis' ],
+	'assets' => [
+		'//extensions/wikia/ArticleVideo/scripts/ooyala/core.js',
+		'//extensions/wikia/ArticleVideo/scripts/ooyala/main_html5.js',
+		'//extensions/wikia/ArticleVideo/scripts/ooyala/html5-skin.js',
+		'//extensions/wikia/ArticleVideo/scripts/ooyala-player.js',
```
Comment (rogatty) [inline extensions/wikia/AssetsManager/config.php]: Can a featured and related video be on the same page? If yes, then we will send users a lot of duplicated code. Maybe we should reorganize it differently, especially that related video code is pretty small compared to the rest.

File: skins/oasis/modules/BodyController.class.php [hunk 8]
```diff
@@ -460,9 +460,21 @@ public function executeIndex() {
 			$this->wg->enableArticleVideo && isset( $this->wg->articleVideoFeaturedVideos[$title] ) &&
 			$this->isFeaturedVideosValid( $this->wg->articleVideoFeaturedVideos[$title] );
 		$this->setVal( 'enableArticleVideo', $enableArticleVideo );
```
Comment (latata) [inline skins/oasis/modules/BodyController.class.php]: I would rename it to `$enableArticleFeaturedVideo`

File: extensions/wikia/AssetsManager/config.php (hunk 7 above)
Comment (latata) [reply extensions/wikia/AssetsManager/config.php]: 👍

File: extensions/wikia/ArticleVideo/scripts/article-related-video.js [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+
+			// these functions return candidate for following DOM element for related video
+			placements = [
+				function () {
+					return $articleContent.children('h2').first().nextUntil('h2', 'p').last();
```
Comment (latata) [inline extensions/wikia/ArticleVideo/scripts/article-related-video.js]: Looks like we could extract this `$articleContent.children('h2')` to variable

File: extensions/wikia/ArticleVideo/scripts/article-related-video.js [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+				},
+				function () {
+					return $articleContent.children('h2').eq(1).nextUntil('h2', 'p').first();
+				},
+				function () {
+					return $articleContent.children('h2').first().nextUntil('h2', 'p').eq(-3);
+				}
+			];
+			
+			$.each(placements, function (i, func) {
```
Comment (latata) [inline extensions/wikia/ArticleVideo/scripts/article-related-video.js]: Can't we use `placements.forEach()`?

File: extensions/wikia/ArticleVideo/scripts/article-related-video.js [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+			
+			if (!$candidate || !$candidate.length) {
+				return -1;
+			}
+
+			$fakeDiv = $('<div>').css({'overflow':'hidden'});
+			candidateWidth = $fakeDiv.insertBefore($candidate).width();
+			$fakeDiv.remove();
+			
+			if (candidateWidth < 500) {
```
Comment (latata) [inline extensions/wikia/ArticleVideo/scripts/article-related-video.js]: Does `500` always fit? Both to large/small screens?

File: extensions/wikia/ArticleVideo/scripts/article-related-video.js [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+				player.mb.subscribe(window.OO.EVENTS.PLAYBACK_READY, 'ui-title-update', function () {
+					var videoTitle = $video.find('.oo-state-screen-title').text();
+
+					$video.find('.video-title').text(videoTitle);
+					$video.show();
+				});
+			}
+		}
+
+		function rateVideoPlacement($candidate, $best) {
```
Comment (latata) [inline extensions/wikia/ArticleVideo/scripts/article-related-video.js]: How about make it more simple and just check if the candidate fits or not. Not to rate them. As there are a lot of cases that we don't take into consideration so there is high possibility that worse rated candidate actually would fit better.

File: extensions/wikia/ArticleVideo/scripts/article-related-video.js [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+					rating = candidateRating;
+				}
+			});
+			
+			if ($followingSibling && $followingSibling.length) {
+				$video = $video.detach();
+				$followingSibling.before( $video );
+
+				player.mb.subscribe(window.OO.EVENTS.PLAYBACK_READY, 'ui-title-update', function () {
+					var videoTitle = $video.find('.oo-state-screen-title').text();
```
Comment (latata) [inline extensions/wikia/ArticleVideo/scripts/article-related-video.js]: I've just realised that we can probably use `player.getTitle()` here. ;)

File: extensions/wikia/ArticleVideo/scripts/article-related-video.js [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+
+				if (candidateRating > rating) {
+					$followingSibling = $followingSiblingCandidate;
+					rating = candidateRating;
+				}
+			});
+			
+			if ($followingSibling && $followingSibling.length) {
+				$video = $video.detach();
+				$followingSibling.before( $video );
```
Comment (latata) [inline extensions/wikia/ArticleVideo/scripts/article-related-video.js]: How about make it oneliner using `insertBefore`. Something like `$video.insertBefore($followingSibling)`

File: extensions/wikia/ArticleVideo/scripts/article-related-video.js (hunk 11 above)
Comment (weronika) [reply extensions/wikia/ArticleVideo/scripts/article-related-video.js]: Yes, I would say that 500 is a minimum, and it's for both small and large screens.

File: extensions/wikia/ArticleVideo/ArticleVideoController.class.php [hunk 15]
```diff
@@ -13,4 +13,22 @@ public function index() {
...
+		$wg = F::app()->wg;
+
+		$title = RequestContext::getMain()->getTitle()->getPrefixedDBkey();
```
Comment (idradm) [inline extensions/wikia/ArticleVideo/ArticleVideoController.class.php]: Current title should be available under global `wgTitle` variable.

File: extensions/wikia/ArticleVideo/scripts/article-related-video.js [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+
+		function rateVideoPlacement($candidate, $best) {
+			var $fakeDiv, bestWidth, candidateWidth;
+			
+			if (!$candidate || !$candidate.length) {
+				return -1;
+			}
+
+			$fakeDiv = $('<div>').css({'overflow':'hidden'});
+			candidateWidth = $fakeDiv.insertBefore($candidate).width();
```
Comment (idradm) [inline extensions/wikia/ArticleVideo/scripts/article-related-video.js]: Are such operations heavy for browser rendering?

File: skins/oasis/modules/BodyController.class.php [hunk 17]
```diff
@@ -456,13 +456,25 @@ public function executeIndex() {
 		$this->setVal( 'displayHeader', !$this->wg->HideNavigationHeaders );
 
 		$title = $this->wg->Title->getPrefixedDBkey();
-		$enableArticleVideo =
-			$this->wg->enableArticleVideo && isset( $this->wg->articleVideoFeaturedVideos[$title] ) &&
+		$enableArticleFeaturedVideo =
+			$this->wg->enableArticleFeaturedVideo && isset( $this->wg->articleVideoFeaturedVideos[$title] ) &&
 			$this->isFeaturedVideosValid( $this->wg->articleVideoFeaturedVideos[$title] );
-		$this->setVal( 'enableArticleVideo', $enableArticleVideo );
+		$this->setVal( 'enableArticleFeaturedVideo', $enableArticleFeaturedVideo );
```
Comment (idradm) [inline skins/oasis/modules/BodyController.class.php]: Why do we need to set those variables in `BodyController`? Is it possible to make those checks inside `ArticleVideo` extension?

File: extensions/wikia/ArticleVideo/ArticleVideo.i18n.php [hunk 18]
```diff
@@ -4,8 +4,10 @@
...
 $messages['en'] = [
 	'articlevideo-watch' => 'watch • $1',
+	'articlevideo-related-video' => 'Related video',
 ];
 
 $messages['qqq'] = [
-	'articlevideo-watch' => '"Watch" string with a video length',
-];
\ No newline at end of file
+	'articlevideo-watch' => '"Watch" string before video time',
```
Comment (latata) [inline extensions/wikia/ArticleVideo/ArticleVideo.i18n.php]: Why did you change this?

File: extensions/wikia/ArticleVideo/scripts/article-related-video.js [hunk 19]
```diff
@@ -0,0 +1,121 @@
...
+				
+				$followingSiblingCandidate = func();
+
+				if (isPlacementBetter($followingSiblingCandidate, $followingSibling)) {
+					$followingSibling = $followingSiblingCandidate;
+				}
+			});
+
+			if ($followingSibling && $followingSibling.length) {
+				$video.detach().insertBefore($followingSibling);
```
Comment (latata) [inline extensions/wikia/ArticleVideo/scripts/article-related-video.js]: I think, we don't need detach here.

File: extensions/wikia/ArticleVideo/scripts/article-related-video.js [hunk 20]
```diff
@@ -0,0 +1,121 @@
...
+					$video.find('.video-title').text(videoTitle);
+					$video.find('.video-time').text(videoTime);
+
+					$video.show();
+				});
+			}
+		}
+
+		function isPlacementBetter($candidate, $currentBest) {
+			var $fakeDiv, bestWidth, candidateWidth;
```
Comment (latata) [inline extensions/wikia/ArticleVideo/scripts/article-related-video.js]: Shouldn't we have each variable in separate line? 
And I'd also set `$fakeDiv` value while declaring it

File: extensions/wikia/ArticleVideo/scripts/article-related-video.js [hunk 21]
```diff
@@ -0,0 +1,121 @@
...
+			
+			if (candidateWidth < 500) {
+				return false;
+			}
+
+			if (!$currentBest || !$currentBest.length) {
+				return true;
+			}
+
+			bestWidth = $fakeDiv.detach().insertBefore($currentBest).width();
```
Comment (latata) [inline extensions/wikia/ArticleVideo/scripts/article-related-video.js]: I think, we don't need detach here.

File: extensions/wikia/ArticleVideo/ArticleVideo.i18n.php [hunk 22]
```diff
@@ -4,8 +4,10 @@
 
 $messages['en'] = [
 	'articlevideo-watch' => 'watch • $1',
+	'articlevideo-related-video' => 'Related video',
```
Comment (latata) [inline extensions/wikia/ArticleVideo/ArticleVideo.i18n.php]: I'd make this parametrized with time, the same as watch label.

File: extensions/wikia/ArticleVideo/scripts/article-related-video.js (hunk 20 above)
Comment (weronika) [reply extensions/wikia/ArticleVideo/scripts/article-related-video.js]: I'll place them in separate lines. But I wouldn't set value right away, it's superfluous before the first if.

File: extensions/wikia/ArticleVideo/ArticleVideo.i18n.php (hunk 22 above)
Comment (weronika) [reply extensions/wikia/ArticleVideo/ArticleVideo.i18n.php]: Why is that? I changed the string for 'watch' also :) I think it's better to set the bullet in css, especially that the duration is out of context of the label and in future this elements may be separated.

Comment (latata) [reply extensions/wikia/ArticleVideo/ArticleVideo.i18n.php]: To make the order customizable by translations. But that's true if we want to split that in the future we should have these separated.

**Review by latata — APPROVED** [review]
LGTM

File: extensions/wikia/ArticleVideo/ArticleVideoController.class.php [hunk 23]
```diff
@@ -1,16 +1,67 @@
...
 
 class ArticleVideoController extends WikiaController {
-	public function index() {
-		$wg = F::app()->wg;
+	public function featured() {
+		global $wgTitle;
 
-		$wg->Out->addModules( 'ext.ArticleVideo' );
+		$wg = $this->getApp()->wg;
+		$title = $wgTitle->getPrefixedDBkey();
```
Comment (idradm) [inline extensions/wikia/ArticleVideo/ArticleVideoController.class.php]: Let's stick to one way of accessing globals. Go with `$wg->Title->`.

File: extensions/wikia/ArticleVideo/ArticleVideo.hooks.php [hunk 24]
```diff
@@ -5,9 +5,26 @@ public static function onBeforePageDisplay( \OutputPage $out, \Skin $skin ) {
...
+			\Wikia::addAssetsToOutput( 'ooyala_scss' );
+			\Wikia::addAssetsToOutput( 'ooyala_js' );
+		}
+
+		if ( $wg->enableArticleFeaturedVideo && isset( $wg->articleVideoFeaturedVideos[$title] ) ) {
+			\Wikia::addAssetsToOutput( 'article_featured_video_scss' );
+			\Wikia::addAssetsToOutput( 'article_featured_video_js' );
+		}
+		
+		if ( $wg->enableArticleRelatedVideo && isset( $relatedVideo ) ) {
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/ArticleVideo.hooks.php]: We check this condition 3 times in this file and then again in the controller. How about extracting it to a method? The same goes for the featured video.

File: extensions/wikia/ArticleVideo/ArticleVideo.hooks.php [hunk 25]
```diff
@@ -17,14 +34,26 @@ public static function onMakeGlobalVariablesScript( array &$vars, OutputPage $ou
...
 				'ooyalaPlayerBrandingId' => $wg->ooyalaApiConfig['playerBrandingId'],
 			];
 			$vars['wgFeaturedVideoId'] = $wg->articleVideoFeaturedVideos[$title]['videoId'];
 		}
 
+		if ( $wg->enableArticleRelatedVideo && isset( $wg->articleVideoRelatedVideos ) ) {
+			$relatedVideo = ArticleVideoController::getRelatedVideoData( $wg->articleVideoRelatedVideos, $title );
+
+			if ( isset( $relatedVideo['videoId'] ) ) {
+				$vars['wgOoyalaParams'] = [
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/ArticleVideo.hooks.php]: If you extract the checks as described in https://github.com/Wikia/app/pull/12529/files#r104894328 then we could set it only once `if ( shoudLoadArticleFeaturedVideo() || shouldLoadArticleRelatedVideo() )`

File: extensions/wikia/ArticleVideo/ArticleVideo.i18n.php (hunk 18 above)
Comment (rogatty) [reply extensions/wikia/ArticleVideo/ArticleVideo.i18n.php]: `'articlevideo-watch' => 'watch • $1'` is untouched, so this description seems to not be true.

File: extensions/wikia/ArticleVideo/ArticleVideo.hooks.php [hunk 26]
```diff
@@ -5,9 +5,26 @@ public static function onBeforePageDisplay( \OutputPage $out, \Skin $skin ) {
...
 
-		if ( isset( $wg->articleVideoFeaturedVideos[$title] ) ) {
-			\Wikia::addAssetsToOutput( 'article_video_scss' );
-			\Wikia::addAssetsToOutput( 'article_video_js' );
+		if ( isset( $wg->articleVideoRelatedVideos ) ) {
+			$relatedVideo = ArticleVideoController::getRelatedVideoData( $wg->articleVideoRelatedVideos, $title );
+		}
+
+		if ( ( $wg->enableArticleFeaturedVideo &&
+		       isset( $wg->articleVideoFeaturedVideos[$title] ) ) ||
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/ArticleVideo.hooks.php]: This is hard to read, please stick to the [guidelines](https://github.com/Wikia/guidelines/blob/master/PHP/CodingConventions.md#conditional-logic).

File: extensions/wikia/ArticleVideo/ArticleVideoController.class.php [hunk 27]
```diff
@@ -1,16 +1,67 @@
...
+	}
+
+	public function related() {
+		global $wgTitle;
+
+		$wg = $this->getApp()->wg;
+		$title = $wgTitle->getPrefixedDBkey();
+
+		$relatedVideo = null;
+		if ( $wg->enableArticleRelatedVideo &&
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/ArticleVideoController.class.php]: Please stick to the [guidelines](https://github.com/Wikia/guidelines/blob/master/PHP/CodingConventions.md#conditional-logic).

File: extensions/wikia/ArticleVideo/ArticleVideoController.class.php [hunk 28]
```diff
@@ -1,16 +1,67 @@
...
+		     isset( $wg->articleVideoRelatedVideos )
+		) {
+			$relatedVideo =
+				self::getRelatedVideoData( $wg->articleVideoRelatedVideos, $title );
+		}
+		$enableArticleRelatedVideo = $relatedVideo && $this->isRelatedVideosValid( $relatedVideo );
+
+		if ( $enableArticleRelatedVideo ) {
+			$this->setVal( 'relatedVideo',
+				self::getRelatedVideoData( $wg->articleVideoRelatedVideos, $title ) );
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/ArticleVideoController.class.php]: Please stick to the [guidelines](https://github.com/Wikia/guidelines/blob/master/PHP/CodingConventions.md#multi-line-statements). So, this would be:
```
$this->setVal( 
  'relatedVideo',
  self::getRelatedVideoData( $wg->articleVideoRelatedVideos, $title )
);
```

File: extensions/wikia/ArticleVideo/scripts/article-related-video.js [hunk 29]
```diff
@@ -0,0 +1,123 @@
...
+				},
+				function () {
+					return $articleHeaders.first().nextUntil('h2', 'p, ul').eq(-3);
+				}
+			];
+
+			placementCallbacks.forEach(function(func) {
+				var $followingSiblingCandidate;
+				
+				$followingSiblingCandidate = func();
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/scripts/article-related-video.js]: This can be merged with the line above.

File: extensions/wikia/ArticleVideo/scripts/article-related-video.js [hunk 30]
```diff
@@ -0,0 +1,123 @@
...
+				bestWidth,
+				candidateWidth;
+
+			if (!$candidate || !$candidate.length) {
+				return false;
+			}
+
+			$fakeDiv = $('<div>').css({'overflow':'hidden'});
+			candidateWidth = $fakeDiv.insertBefore($candidate).width();
+			$fakeDiv.remove();
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/scripts/article-related-video.js]: Doesn't it cause any content jumps?

File: extensions/wikia/ArticleVideo/scripts/article-related-video.js [hunk 31]
```diff
@@ -0,0 +1,123 @@
...
+			// these callbacks return candidate for following DOM element for related video
+			placementCallbacks = [
+				function () {
+					return $articleHeaders.first().nextUntil('h2', 'p, ul').last();
+				},
+				function () {
+					return $articleHeaders.eq(1).nextUntil('h2', 'p, ul').first();
+				},
+				function () {
+					return $articleHeaders.first().nextUntil('h2', 'p, ul').eq(-3);
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/scripts/article-related-video.js]: Please add comments explaining why we picked these selectors.

File: extensions/wikia/ArticleVideo/scripts/article-related-video.js (hunk 11 above)
Comment (rogatty) [reply extensions/wikia/ArticleVideo/scripts/article-related-video.js]: Please explain in a comment why we picked 500px.

File: extensions/wikia/ArticleVideo/scripts/article-related-video.js [hunk 32]
```diff
@@ -0,0 +1,123 @@
...
+			
+			if (candidateWidth < 500) {
+				return false;
+			}
+
+			if (!$currentBest || !$currentBest.length) {
+				return true;
+			}
+
+			bestWidth = $fakeDiv.insertBefore($currentBest).width();
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/scripts/article-related-video.js]: Can we do `bestWidth = candidateWidth` in the `if` below instead of modifying the DOM twice?

File: extensions/wikia/ArticleVideo/scripts/article-related-video.js [hunk 33]
```diff
@@ -0,0 +1,123 @@
...
+
+			player.mb.subscribe(OO.EVENTS.PAUSED, 'related-video', function () {
+				track({
+					action: tracker.ACTIONS.CLICK,
+					label: 'related-video-paused'
+				});
+			});
+
+			track({
+				action: tracker.ACTIONS.IMPRESSION,
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/scripts/article-related-video.js]: Shouldn't we do this after `$video.show();`? I think now we're tracking the impression even if there is no good place to put the video in.

Comment (latata) [reply extensions/wikia/ArticleVideo/scripts/article-related-video.js]: 👍

File: extensions/wikia/ArticleVideo/scripts/article-related-video.js (hunk 30 above)
Comment (weronika) [reply extensions/wikia/ArticleVideo/scripts/article-related-video.js]: it does not - I tested

**Review by rogatty — APPROVED** [review]

File: extensions/wikia/ArticleVideo/ArticleVideoController.class.php [hunk 34]
```diff
@@ -1,16 +1,49 @@
...
+	}
 
-		$wg->Out->addModules( 'ext.ArticleVideo' );
+	public function related() {
+		$wg = $this->getApp()->wg;
+		$title = $wg->Title->getPrefixedDBkey();
 
-		$title = RequestContext::getMain()->getTitle()->getPrefixedDBkey();
+		$relatedVideo = self::getRelatedVideoData( $wg->articleVideoRelatedVideos, $title );
+		$enableArticleRelatedVideo = ArticleVideoHooks::isRelatedVideoEmbedded( $relatedVideo );;
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/ArticleVideoController.class.php]: Double semicolon.
